tests/run-tests.py

```diff
@@ -896,22 +877,93 @@ def run_one_test(test_file):
             skipped_tests.append(test_name)
             return
 
-        testcase_count.add(len(output_expected.splitlines()))
+        # Look at the output of the test to see if unittest was used.
+        uses_unittest = False
+        output_mupy_lines = output_mupy.splitlines()
+        if any(
+            line == b"ImportError: no module named 'unittest'" for line in output_mupy_lines[:3]
+        ):
+            print("error: test {} requires unittest".format(test_file))
+            print("(eg run `mpremote mip install unittest` to install it)")
+            sys.exit(1)
```

Comment:
This can cause the process to hang because the thread pool isn't shut down. From a quick search there's no simple and straightforward way to do it apart from `os._exit()`, which does work for me.

Reply:
Good catch, thanks!  Indeed it can hang.

I've changed it to raise an exception, which is caught outside the thread poll runner.  Tested and it works (no longer hangs).

Reply:
Ok, works.